### PR TITLE
#43 added player death event

### LINKED ISSUE #43: Create Event/Death
Create an event for player death that is triggered when player dies

### PR DESCRIPTION
**Pull request by Swackles**
fixes #43 
Creates a player death event that gets called when the player dies.
Also adds in a console command for killing the player

### CODE REVIEW

File: Assets/Scripts/Entity/Player.cs [hunk 1]
```diff
@@ -45,11 +48,18 @@ public void Hurt(Vector2 KnockBack)
...
+                Kill();
             } else 
                 RB.AddForce(KnockBack);
-        }  
+        }
+
+        [Command("Kill")]
+        public void Kill()
+        {
+            EventSystem.Current.PlayerDeath(this);
```
Comment (tlepiko) [inline Assets/Scripts/Entity/Player.cs:59]: Object reference not set to an instance of an object.

File: Assets/Scripts/EventSystems/EventSystem.cs [hunk 2]
```diff
@@ -12,8 +13,25 @@ private void Awake()
         {
             Current = this;
             SubscribeMedicine();
+            SubscribeDeath();
         }
 
+        #region Deadth
```
Comment (tlepiko) [inline Assets/Scripts/EventSystems/EventSystem.cs:19]: Spelling mistake? Not sure if relevant.

File: Assets/Scripts/Entity/Player.cs (hunk 1 above)
Comment (tlepiko) [reply Assets/Scripts/Entity/Player.cs:59]: Problem related to Eventsystem.

**Review by tlepiko — CHANGES_REQUESTED** [review]
Eventsystem Current results in a Reference not set to an instance of an object.

Spelling mistake in Eventsystem "Region Deadth".

File: Assets/Scripts/Entity/Player.cs (hunk 1 above)
Comment (Swackles) [reply Assets/Scripts/Entity/Player.cs:59]: This is cause EventSystem does not exist in that scene

**Review by tlepiko — APPROVED** [review]

**Pull request merged**